app/plugin/FeatureInfoWindow.js

```diff
@@ -0,0 +1,136 @@
+Ext.define('CpsiMapview.plugin.FeatureInfoWindow', {
+    extend: 'Ext.plugin.Abstract',
+    alias: 'plugin.cmv_feature_info_window',
+    pluginId: 'cmv_feature_info_window',
+
+    requires: [
+        'Ext.Component',
+        'CpsiMapview.view.window.MinimizableWindow',
+        'BasiGX.view.grid.FeaturePropertyGrid',
+        'BasiGX.util.Map'
+    ],
+
+    init: function () {
+        var me = this;
+
+        var mapComp = BasiGX.util.Map.getMapComponent();
+        var map = mapComp.getMap();
+
+        map.on('singleclick', function (evt) {
+            me.requestFeatureInfos(mapComp, evt);
+        });
+    },
+
+    requestFeatureInfos: function (mapComp, evt) {
+        var me = this;
+
+        var map = mapComp.getMap();
+
+        var layers = [];
+        var resolution = map.getView().getResolution();
+        var projection = map.getView().getProjection();
+
+        var format = new ol.format.GeoJSON();
+
+        map.forEachLayerAtPixel(evt.pixel, function (layer) {
+            layers.push(layer);
+        }, undefined, function (layer) {
+            return layer.get('featureInfoWindow');
+        });
+
+        var window = me.openFeatureInfoWindow(map, evt);
+
+        if (layers.length <= 1) {
+            window.setLayout('fit');
+        } else {
+            window.setLayout({
+                type: 'accordion',
+                titleCollapse: false,
+                animate: true
+            });
+        }
+
+        if (layers.length === 0) {
+            window.add(Ext.create('Ext.Component', {
+                html: '<span>No results found</span>'
+            }));
+        } else {
+            layers.forEach(function (layer) {
+                var url = layer.getSource().getGetFeatureInfoUrl(evt.coordinate, resolution, projection, {
+                    INFO_FORMAT: 'geojson'
+                });
+
+                Ext.Ajax.request({
+                    url: url
+                }).then(function (response) {
+                    var features = format.readFeatures(response.responseText);
+                    window.add(me.createFeaturePanels(mapComp, layer, features));
+                }).then(undefined, function (error) {
+                    console.error(error);
```

Comment:
Use the `Ext.log` function here? Some WMS layers could return XML errors which it would be useful to log.

Reply:
I put it in there. I just noticed that ExtJs seems to have the bad habit of not reporting any exceptions raised inside of promises so I put the manual error reporting there.